### PR TITLE
[spscqueue] Update to version 1.0

### PR DESCRIPTION
**Pull request by rigtorp**
- Updates to latest upstream version 1.0
- New version adds proper support for `find_package`

### CODE REVIEW

File: ports/spscqueue/portfile.cmake [hunk 1]
```diff
@@ -5,15 +5,22 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
```
Comment (NancyLi1013) [inline ports/spscqueue/portfile.cmake:5]: `include(vcpkg_common_functions)` is no longer needed.
Could you please this line?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/spscqueue/portfile.cmake [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
 )
 
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SPSCQueue)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib)
```
Comment (JackBoosY) [inline ports/spscqueue/portfile.cmake:18]: Is this a header only library? Why do you remove this folder?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/spscqueue/portfile.cmake (hunk 2 above)
Comment (rigtorp) [reply ports/spscqueue/portfile.cmake:18]: Yes it's created by the cmake export file lib/cmake/SPSCQueueConfig.cmake which is moved by `vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SPSCQueue)` to where vcpkg wants the export files. But the `lib/` dir is left empty and needs to be removed.

Comment (rigtorp) [reply ports/spscqueue/portfile.cmake:18]: @JackBoosY any other questions preventing merging?

**Review by NancyLi1013 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**